dangerfile.js

```diff
@@ -0,0 +1,120 @@
+	if (bytes === 0) return '0 Bytes';
+
+	const k = 1000;
+	const dm = decimals < 0 ? 0 : decimals;
+	const sizes = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+
+	const i = Math.floor(Math.log(Math.abs(bytes)) / Math.log(k));
+
+	return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i];
+}
+
+const maybePlus = (from, to) => from < to ? "+" : "";
+
+const absDiff = (from, to) => {
+	if (from === to) {
+		return formatBytes(0);
+	}
+
+	return `${maybePlus(from, to)}${formatBytes(to - from)}`;
+}
+
+const percDiff =  (from, to) => {
+	if (from === to) {
+		return '0%';
+	}
+
+	return `${maybePlus(from, to)}${
+		(100 * (to - from) / (from || to)).toFixed(1)
+	}%`;
+}
+
+const getSummary = (rows, totalMasterFileSize, totalFileSize) => {
+	const numFiles = rows.length;
+	const maybeS = rows.length > 0 ? 's' : '';
+	const byteDiff = absDiff(totalMasterFileSize, totalFileSize);
+	const percentDiff = percDiff(totalMasterFileSize, totalFileSize);
+
+	return `A total of ${numFiles} file${maybeS} have changed, with a combined diff of ${byteDiff} (${percentDiff}).`;
+}
+
+const handleRenamedError = (file, type) => async err => {
+	// Renamed files show up in modified but don't exist in master,
+	// so this is the only error here we want to handle.
+	if (type === 'modified' && err.message.includes("exists on disk, but not in 'master'")) {
+		return '';
+	}
+
+	// Otherwise, rethrow, cuz we haven't handled any other errors.
+	throw err;
+}
+
+const getChangedMinifiedFiles = async () => {
+	const result = await git.diff(['--name-only', '--no-renames', 'master...']);
+
+	return result
+		? result.split(/\r?\n/g).filter(file => file.includes('.min.js'))
+		: [];
+};
+
+const run = async () => {
+	// Check if master exists & check it out if not.
+	const result = await git.branch(['--list', 'master']);
+	if (result.all.length === 0) {
+		await git.branch(['master', 'origin/master']);
+	}
+
+	const minified = await getChangedMinifiedFiles();
+
+	if (minified.length === 0) {
+		markdown(`## No JS Changes`);
+		return;
+	}
+
+	const rows = [];
+	let totalFileSize = 0;
+	let totalMasterFileSize = 0;
+
+	for (const file of minified) {
+		const [fileContents, fileMasterContents] = await Promise.all([
+			fs.readFile(file, 'utf-8').catch(() => ''),
+			git.show([`master:${file}`]).catch(() => ''),
+		]);
+
+		const [fileSize, fileMasterSize] = await Promise.all([
+			gzipSize(fileContents),
+			gzipSize(fileMasterContents),
+		]);
+
+		totalFileSize += fileSize;
+		totalMasterFileSize +=fileMasterSize
+
+		rows.push([
+			file,
+			formatBytes(fileMasterSize),
+			formatBytes(fileSize),
+			absDiff(fileMasterSize, fileSize),
+			percDiff(fileMasterSize, fileSize),
+		]);
+	}
+
+	markdown(`${getSummary(rows, totalMasterFileSize, totalFileSize)}
+
+## JS File Size Changes (gzipped)
+
+| file | master | pull | size diff | % diff |
+| --- | --- | --- | --- | --- |
+${rows.map(row => `| ${row.join(' | ')} |`).join('\n')}`);
```

Comment:
1. The summary is of the changed JS files, so it should be under that header.
1. I really think that the full list of changed files should be collapsed by default. This list can be really long for some (rare) PRs and will only contain one item for most (in which case the summary already contains the size change).
```suggestion
	markdown(`## JS File Size Changes (gzipped)

${getSummary(rows, totalMasterFileSize, totalFileSize)}

<details>

| file | master | pull | size diff | % diff |
| --- | --- | --- | --- | --- |
${rows.map(row => `| ${row.join(' | ')} |`).join('\n')}

</details>
`);
```